BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -519,6 +554,146 @@ InitSmbiosStringPtr (
   ASSERT_EFI_ERROR (Status);
 }
 
+/**
+  This function builds required processor info SMBIOS type.
+**/
+EFI_STATUS
+BuildProcessorInfo (
+  VOID
+  )
+{
+  SMBIOS_TABLE_TYPE4              SmbiosRecord;
+  UINT32                          Eax01;
+  UINT32                          Ebx01;
+  UINT32                          Ecx01;
+  UINT32                          Edx01;
+  PROCESSOR_ID_DATA               ProcessorId;
+  UINT32                          ProcessorVoltage;
+  UINTN                           NumberOfThreads;
+  UINTN                           NumberOfCores;
+  EFI_STATUS                      Status;
+  SYS_CPU_INFO                   *SysCpuInfo;
+  UINTN                           Idx;
+  UINTN                           BufIdx;
+  CHAR8                           CpuBrandBuf[64];
+  BOOLEAN                         HyperThreadEnable;
+  UINT16                          Family;
+  UINT16                          BclkFrequency;
+
+  for (Idx = 0, BufIdx = 0; Idx < 3; Idx++) {
+    AsmCpuid (
+              (UINT32)(CPUID_BRAND_STRING1 + Idx),
+              (UINT32 *)&CpuBrandBuf[BufIdx],
+              (UINT32 *)&CpuBrandBuf[BufIdx+4],
+              (UINT32 *)&CpuBrandBuf[BufIdx+8],
+              (UINT32 *)&CpuBrandBuf[BufIdx+12]
+             );
+    BufIdx += 16;
+  }
+  if (CpuBrandBuf[0] == 0) {
+    AsciiStrCpyS (CpuBrandBuf, sizeof(CpuBrandBuf), BRAND_STRING_UNSUPPORTED);
+  }
+  CpuBrandBuf[48] = 0;
```

Comment:
Why are we restricting the string to only 48 bytes?

Reply:
CPUID can return 16 bytes for one cycle. We only do 3 cycles, so it should be less than 16*3.